skyportal/handlers/api/comment.py

```diff
@@ -83,19 +87,27 @@ def post(self):
         obj_id = data['obj_id']
         # Ensure user/token has access to parent source
         _ = Source.get_if_owned_by(obj_id, self.current_user)
+        user_group_ids = [g.id for g in self.current_user.groups]
+        group_ids = data.pop("group_ids", user_group_ids)
+        if group_ids == []:
+            group_ids = user_group_ids
+        group_ids = [gid for gid in group_ids if gid in user_group_ids]
```

Comment:
I quite like using sets for this sort of thing, but what you have is perfectly valid.

```
group_ids = set(group_ids).intersect(user_group_ids)
```